common/lib/utils.cpp

```diff
@@ -194,10 +194,24 @@ double boxMuller(float m, float s) {
 
 double randf() { return (double)random() / RAND_MAX; }
 
-char *vstrcatf(char **dest, const char *fmt, va_list ap) {
+char *vstrcatf(char **dest, const char *fmt, va_list &ap) {
```

Comment:
I'm surprised you don't need `const char* restrict fmt`.

Reply:
To my knowledge `restrict` is a C not C++ keyword?

Are you concerned about aliasing between the `*dest` and `fmt` pointers?

I dont fully understand, why we might need it..